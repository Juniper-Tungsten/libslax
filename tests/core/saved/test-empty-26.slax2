version 1.2;

param $min = 4;
param $max = 8;

main <top> {
    for-each (1 ... 10) {
        <up> .;
    }
    
    for-each (10 ... 1) {
        <down> .;
    }
    
    for $x ($min ... $max) {
        <up> $x;
    }
    
    for $y ($max ... $min) {
        <down> $y;
    }
}
