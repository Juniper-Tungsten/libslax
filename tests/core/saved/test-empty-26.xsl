<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="slax">
  <xsl:param name="min" select="4"/>
  <xsl:param name="max" select="8"/>
  <xsl:template match="/">
    <top>
      <xsl:for-each xmlns:slax="http://xml.libslax.org/slax" select="slax:build-sequence(1, 10)">
        <up>
          <xsl:value-of select="."/>
        </up>
      </xsl:for-each>
      <xsl:for-each xmlns:slax="http://xml.libslax.org/slax" select="slax:build-sequence(10, 1)">
        <down>
          <xsl:value-of select="."/>
        </down>
      </xsl:for-each>
      <xsl:variable name="slax-dot-1" select="."/>
      <xsl:for-each xmlns:slax="http://xml.libslax.org/slax" select="slax:build-sequence($min, $max)">
        <xsl:variable name="x" select="."/>
        <xsl:for-each select="$slax-dot-1">
          <up>
            <xsl:value-of select="$x"/>
          </up>
        </xsl:for-each>
      </xsl:for-each>
      <xsl:variable name="slax-dot-2" select="."/>
      <xsl:for-each xmlns:slax="http://xml.libslax.org/slax" select="slax:build-sequence($max, $min)">
        <xsl:variable name="y" select="."/>
        <xsl:for-each select="$slax-dot-2">
          <down>
            <xsl:value-of select="$y"/>
          </down>
        </xsl:for-each>
      </xsl:for-each>
    </top>
  </xsl:template>
</xsl:stylesheet>
