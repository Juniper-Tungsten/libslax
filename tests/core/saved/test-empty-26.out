<?xml version="1.0"?>
<top>
  <up>1</up>
  <up>2</up>
  <up>3</up>
  <up>4</up>
  <up>5</up>
  <up>6</up>
  <up>7</up>
  <up>8</up>
  <up>9</up>
  <up>10</up>
  <down>10</down>
  <down>9</down>
  <down>8</down>
  <down>7</down>
  <down>6</down>
  <down>5</down>
  <down>4</down>
  <down>3</down>
  <down>2</down>
  <down>1</down>
  <up>4</up>
  <up>5</up>
  <up>6</up>
  <up>7</up>
  <up>8</up>
  <down>8</down>
  <down>7</down>
  <down>6</down>
  <down>5</down>
  <down>4</down>
</top>
